### PR TITLE
fix(doc): fix kube client interface doc.

### PR DESCRIPTION
**Pull request by longkai**
For the interface at `pkg/kube/interface.go`:

- add missing interface methods doc.
- fix some wrong english grammar.

**What this PR does / why we need it**:

Fix kube client interface doc.

**If applicable**:
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
Note, this requires the approval of a second maintainer before merging.

**Review by cndoit18 — APPROVED** [review]

**Review by technosophos — APPROVED** [review]
LGTM

**Pull request merged**
